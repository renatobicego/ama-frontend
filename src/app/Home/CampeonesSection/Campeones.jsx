"use client"
import { sliceIntoChunks} from "@/app/utils/utils"
import { Carousel, Typography} from "@/app/utils/materialTailwind"
import ContainerCards from "./ContainerCards"
import useWindowSize from "@/app/utils/hooks/useWindowSize"
import { useEffect, useState } from "react"
import axios from "axios"

const mockedCampeones = [
    {
        name: 'Renzo Cremaschi',
        imgHref: '/imgs/renzo.jpg',
        pruebas: ['110 C/V', '200m']
    },
    {
        name: 'Renata Zanata',
        imgHref: '/imgs/renata.jpeg',
        pruebas: ['5000m Marcha']
    },
    {
        name: 'Ignacio Erario',
        imgHref: '/imgs/erario.jpeg',
        pruebas: ['5000m', '10000m']
    },
    {
        name: 'Alma Marcha',
        imgHref: '/imgs/almaMarcha.jpeg',
        pruebas: ['5000m Marcha']
    },
]

const Campeones = () => {

    const windowSize = useWindowSize()
    const [campeones, setCampeones] = useState([])

    useEffect(async() => {
        const res = await axios.get(`${process.env.NEXT_PUBLIC_URL_API}/campeones`)
        console.log(res.data)
    }, [])

    // Check for how many cards to render in div
    const isMobile = windowSize.width < 768 
    const isDesktop = windowSize.width > 1280

    return(
        <section className="w-full bg-hamburguer-menu-bg py-14">
            <div className="size-section flex justify-between flex-col items-center">
                <Typography
                  className='font-title font-semibold text-white text-2xl lg:text-3xl text-center'
                >
                    Nuestros Campeones Nacionales
                </Typography>
                <Carousel
                    autoplay={true}
                    loop={true}
                    className="md:mt-2 lg:mt-4"
                    navigation={({ setActiveIndex, activeIndex, length }) => (
                        <div className="absolute bottom-0 left-2/4 z-10 flex -translate-x-2/4 gap-2">
                          {new Array(length).fill("").map((_, i) => (
                            <span
                              key={i}
                              className={`block h-1 cursor-pointer rounded-2xl transition-all content-[''] w-12 lg:w-16 ${
                                activeIndex === i ? "bg-white " : "bg-white/50 "
                              }`}
                              onClick={() => setActiveIndex(i)}
                            />
                          ))}
                        </div>
                      )}
                >
                    {/* Render 1 card if mobile, 2 if tablet and 3 if desktop */}
                    {
                        sliceIntoChunks(mockedCampeones, 
                            isMobile ? 1 : (isDesktop ? 3 : 2))
                            .map((chunk, i) => <ContainerCards campeones={chunk} key={i}/>)
                    }
                    
                </Carousel>
            </div>
        </section>
    )
}

export default Campeones