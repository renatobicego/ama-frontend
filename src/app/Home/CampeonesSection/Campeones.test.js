import {cleanup, render, screen} from "@testing-library/react"
import '@testing-library/jest-dom/extend-expect'
import Campeones from "./Campeones"


describe('<Campeones />', () => {

    let campeones 

    // Render component
    beforeEach(() => {
        campeones = render(<Campeones/>)
    })

    afterEach(() => {
        cleanup()
    })

    test('renders content', () => {
        expect(campeones.container).toMatchSnapshot()

        // Check title
        const title = screen.getByText('Nuestros Campeones Nacionales')
        expect(title).toBeInTheDocument()

    })
})