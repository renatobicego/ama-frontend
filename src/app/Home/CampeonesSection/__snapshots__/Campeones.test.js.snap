// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Campeones /> renders content 1`] = `
<div>
  <section
    class="w-full bg-hamburguer-menu-bg py-14"
  >
    <div
      class="size-section flex justify-between flex-col items-center"
    >
      <p
        class="block antialiased font-title font-semibold text-white text-2xl lg:text-3xl text-center"
      >
        Nuestros Campeones Nacionales
      </p>
      <div
        class="relative w-full h-full overflow-x-hidden flex md:mt-2 lg:mt-4"
      >
        <div
          class="w-full h-full inline-block flex-none"
          style="left: 0%; right: 0%; transform: none;"
        >
          <div
            class="w-full flex justify-evenly py-6 px-16"
          >
            <div
              class="relative flex flex-col bg-clip-border rounded-xl bg-white text-gray-700 shadow-md mt-6 w-95% md:w-[45%] lg:w-[30%] overflow-hidden"
            >
              <div
                class="bg-clip-border overflow-hidden bg-white text-gray-700 relative w-full m-0 rounded-none"
              >
                <img
                  alt="Foto de Renzo Cremaschi"
                  class="h-44 md:h-52 lg:h-64 object-cover"
                  data-nimg="1"
                  decoding="async"
                  height="700"
                  loading="lazy"
                  src="/_next/image?url=%2Fimgs%2Frenzo.jpg&w=1920&q=75"
                  srcset="/_next/image?url=%2Fimgs%2Frenzo.jpg&w=750&q=75 1x, /_next/image?url=%2Fimgs%2Frenzo.jpg&w=1920&q=75 2x"
                  style="color: transparent;"
                  width="700"
                />
              </div>
              <div
                class="p-6"
              >
                <h5
                  class="block antialiased tracking-normal text-xl font-semibold leading-snug mb-2 text-title font-title"
                >
                  Renzo Cremaschi
                </h5>
                <p
                  class="block antialiased text-base font-light leading-relaxed text-title font-title"
                >
                  110 C/V / 
                  200m / 
                </p>
              </div>
            </div>
            <div
              class="relative flex flex-col bg-clip-border rounded-xl bg-white text-gray-700 shadow-md mt-6 w-95% md:w-[45%] lg:w-[30%] overflow-hidden"
            >
              <div
                class="bg-clip-border overflow-hidden bg-white text-gray-700 relative w-full m-0 rounded-none"
              >
                <img
                  alt="Foto de Renata Zanata"
                  class="h-44 md:h-52 lg:h-64 object-cover"
                  data-nimg="1"
                  decoding="async"
                  height="700"
                  loading="lazy"
                  src="/_next/image?url=%2Fimgs%2Frenata.jpeg&w=1920&q=75"
                  srcset="/_next/image?url=%2Fimgs%2Frenata.jpeg&w=750&q=75 1x, /_next/image?url=%2Fimgs%2Frenata.jpeg&w=1920&q=75 2x"
                  style="color: transparent;"
                  width="700"
                />
              </div>
              <div
                class="p-6"
              >
                <h5
                  class="block antialiased tracking-normal text-xl font-semibold leading-snug mb-2 text-title font-title"
                >
                  Renata Zanata
                </h5>
                <p
                  class="block antialiased text-base font-light leading-relaxed text-title font-title"
                >
                  5000m Marcha / 
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          class="w-full h-full inline-block flex-none"
          style="left: 100%; right: 100%; transform: none;"
        >
          <div
            class="w-full flex justify-evenly py-6 px-16"
          >
            <div
              class="relative flex flex-col bg-clip-border rounded-xl bg-white text-gray-700 shadow-md mt-6 w-95% md:w-[45%] lg:w-[30%] overflow-hidden"
            >
              <div
                class="bg-clip-border overflow-hidden bg-white text-gray-700 relative w-full m-0 rounded-none"
              >
                <img
                  alt="Foto de Ignacio Erario"
                  class="h-44 md:h-52 lg:h-64 object-cover"
                  data-nimg="1"
                  decoding="async"
                  height="700"
                  loading="lazy"
                  src="/_next/image?url=%2Fimgs%2Ferario.jpeg&w=1920&q=75"
                  srcset="/_next/image?url=%2Fimgs%2Ferario.jpeg&w=750&q=75 1x, /_next/image?url=%2Fimgs%2Ferario.jpeg&w=1920&q=75 2x"
                  style="color: transparent;"
                  width="700"
                />
              </div>
              <div
                class="p-6"
              >
                <h5
                  class="block antialiased tracking-normal text-xl font-semibold leading-snug mb-2 text-title font-title"
                >
                  Ignacio Erario
                </h5>
                <p
                  class="block antialiased text-base font-light leading-relaxed text-title font-title"
                >
                  5000m / 
                  10000m / 
                </p>
              </div>
            </div>
            <div
              class="relative flex flex-col bg-clip-border rounded-xl bg-white text-gray-700 shadow-md mt-6 w-95% md:w-[45%] lg:w-[30%] overflow-hidden"
            >
              <div
                class="bg-clip-border overflow-hidden bg-white text-gray-700 relative w-full m-0 rounded-none"
              >
                <img
                  alt="Foto de Alma Marcha"
                  class="h-44 md:h-52 lg:h-64 object-cover"
                  data-nimg="1"
                  decoding="async"
                  height="700"
                  loading="lazy"
                  src="/_next/image?url=%2Fimgs%2FalmaMarcha.jpeg&w=1920&q=75"
                  srcset="/_next/image?url=%2Fimgs%2FalmaMarcha.jpeg&w=750&q=75 1x, /_next/image?url=%2Fimgs%2FalmaMarcha.jpeg&w=1920&q=75 2x"
                  style="color: transparent;"
                  width="700"
                />
              </div>
              <div
                class="p-6"
              >
                <h5
                  class="block antialiased tracking-normal text-xl font-semibold leading-snug mb-2 text-title font-title"
                >
                  Alma Marcha
                </h5>
                <p
                  class="block antialiased text-base font-light leading-relaxed text-title font-title"
                >
                  5000m Marcha / 
                </p>
              </div>
            </div>
          </div>
        </div>
        <button
          class="!absolute top-2/4 left-4 -translate-y-2/4 rounded-full select-none transition-all disabled:opacity-50 disabled:shadow-none disabled:pointer-events-none w-12 max-w-[48px] h-12 max-h-[48px] text-white hover:bg-white/10 active:bg-white/30 grid place-items-center"
        >
          <svg
            aria-hidden="true"
            class="-ml-1 h-7 w-7"
            fill="none"
            stroke="currentColor"
            stroke-width="3"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M15.75 19.5L8.25 12l7.5-7.5"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
        </button>
        <button
          class="!absolute top-2/4 right-4 -translate-y-2/4 rounded-full select-none transition-all disabled:opacity-50 disabled:shadow-none disabled:pointer-events-none w-12 max-w-[48px] h-12 max-h-[48px] text-white hover:bg-white/10 active:bg-white/30 grid place-items-center"
        >
          <svg
            aria-hidden="true"
            class="ml-1 h-7 w-7"
            fill="none"
            stroke="currentColor"
            stroke-width="3"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8.25 4.5l7.5 7.5-7.5 7.5"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
        </button>
        <div
          class="absolute bottom-0 left-2/4 z-10 flex -translate-x-2/4 gap-2"
        >
          <span
            class="block h-1 cursor-pointer rounded-2xl transition-all content-[''] w-12 lg:w-16 bg-white "
          />
          <span
            class="block h-1 cursor-pointer rounded-2xl transition-all content-[''] w-12 lg:w-16 bg-white/50 "
          />
        </div>
      </div>
    </div>
  </section>
</div>
`;
